cairo/tests/utils/args_gen.py

```diff
@@ -300,6 +348,12 @@ def _gen_arg(
         segments.load_data(base, [dict_ptr, current_ptr])
         return base
 
+    if arg_type_origin is Stack:
+        # Cast to Stack to enable gen_arg method if not already a Stack instance
+        if not isinstance(arg, Stack):
+            arg = Stack(arg)
+        return arg.gen_arg(dict_manager, segments)
```

Comment:
cf above with protocol, this should be called first and be like
```suggestion
    if arg_type_origin is IntoBaseType:
       arg_type, arg = arg.to_base_type()
```